### PR TITLE
Craftable Bluespace Emags!

### PR DESCRIPTION
**Pull request by LucyGrind**
## About The Pull Request
Makes the admin-only bluespace emag craftable by the crew using a bluespace anomoly core. Also adds the RPED beam effect to it.
## Why It's Good For The Game
More opo bluespace core interactions for le ultimate win.
Jacquerel asked for it, don't yell at me if you don't like it :(
Ranged emags are cooler then non-ranged emags, think of all the shenanigans you can do with it!
Providing traitors with more diverse tools can lead to a wider range of antagonist strategies, making it harder for the crew to predict and counteract traitor actions. This can increase the tension and excitement of traitor rounds.
## Changelog
:cl:
add: Craftable bluespace emag! (use anomaly core on emag)
/:cl:

### CODE REVIEW

**Comment by Time-Green** [comment]
Why is it good for the game?????????

**Comment by the-orange-cow** [comment]
Interesting, but I can't imagine wasting a bluespace core on an emag of all things.

**Comment by LucyGrind** [comment]
> Why is it good for the game?????????

read the Why it is good for the game, i expanded on it

**Comment by CCC23** [comment]
the crew wont be able to handle LONG EMAG

**Comment by SomeoneYouProbablyKnow** [comment]
This is nice, a craft that doesn't require rnd to do shit. That means as a traitor you can pivot to it if you see a convenient anomaly floating around instead of hinging your plan on the competency of purple crayon eaters.

**Comment by Loafin34** [comment]
does this show in the crafting menu?

**Comment by LucyGrind** [comment]
> does this show in the crafting menu?

nah its not a menu craftable, its an item interaction between bluespace cores & emags

File: code/game/objects/items/emags.dm [hunk 1]
```diff
@@ -64,8 +65,21 @@
...
 /obj/item/card/emag/ranged_interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	if (can_emag(interacting_with, user))
+		if(pshoom_2 && prob(1))
+			playsound(src, pshoom_2, 40, TRUE)
+		else
+			playsound(src, pshoom, 40, TRUE)
+		user.Beam(interacting_with, icon_state = "rped_upgrade", time = 0.5 SECONDS)
 	return prox_check ? NONE : interact_with_atom(interacting_with, user)
 
+/obj/item/card/emag/item_interaction(mob/living/user, obj/item/tool)
```
Comment (carlarctg) [inline code/game/objects/items/emags.dm:76]: reee use slapcrafting reeee

**Review by Wallemations — APPROVED** [review]

File: code/game/objects/items/emags.dm (hunk 1 above)
Comment (LucyGrind) [reply code/game/objects/items/emags.dm:76]: what the hell is slapcrafting

**Comment by HoodMilk** [comment]
I've messed around with them on a solo server in the past and bluespace emags are hilariously OP. You can use a camera console and hack borgs while they are getting, randomly explode every single light on the station with 0 risk to yourself, hack arcade games WHILE PEOPLE ARE PLAYING THEM to kill them, hijack the emergency shuttle without even entering the cockpit, set air alarms to siphon and much more.

 It's absolutely wild what you can do with a bluespace emag and there's a really good reason they are an admin only item.

**Comment by LucyGrind** [comment]
> I've messed around with them on a solo server in the past and bluespace emags are hilariously OP. You can use a camera console and hack borgs while they are getting repaired or new modules, randomly explode every single light on the station with 0 risk to yourself, hack arcade games WHILE PEOPLE ARE PLAYING THEM to kill them, hijack the emergency shuttle without even entering the cockpit, set air alarms to siphon and much more.
> 
> It's absolutely wild what you can do with a bluespace emag and there's a really good reason they are an admin only item.

shhhhhhhh let it be merged even if for only a moment, it'll be funky af, plus it requires a bluespace core so no harm no foul... i mean you can literally bagulo

File: code/game/objects/items/emags.dm (hunk 1 above)
Comment (LucyGrind) [reply code/game/objects/items/emags.dm:76]: god no nobody uses it

File: code/game/objects/items/emags.dm [hunk 2]
```diff
@@ -64,8 +65,21 @@
...
+	if (istype(tool, /obj/item/assembly/signaler/anomaly/bluespace))
+		qdel(src)
+		qdel(tool)
+		user.put_in_active_hand(new /obj/item/card/emag/bluespace)
+	else return NONE
```
Comment (Mothblocks) [inline code/game/objects/items/emags.dm:81]: New line

File: code/game/objects/items/emags.dm (hunk 1 above)
Comment (Mothblocks) [inline code/game/objects/items/emags.dm:76]: What's the return value of this? I'm noticing one branch returns NONE and another nothing? That's weird

File: code/game/objects/items/emags.dm [hunk 3]
```diff
@@ -64,8 +65,21 @@
...
+		if(pshoom_2 && prob(1))
+			playsound(src, pshoom_2, 40, TRUE)
+		else
+			playsound(src, pshoom, 40, TRUE)
+		user.Beam(interacting_with, icon_state = "rped_upgrade", time = 0.5 SECONDS)
 	return prox_check ? NONE : interact_with_atom(interacting_with, user)
 
+/obj/item/card/emag/item_interaction(mob/living/user, obj/item/tool)
+	if (istype(tool, /obj/item/assembly/signaler/anomaly/bluespace))
+		qdel(src)
```
Comment (Mothblocks) [inline code/game/objects/items/emags.dm:78]: Move qdel(src) to the end--I think that once a thing is qdel'd nothing should be doing anything with it anymore.

File: code/game/objects/items/emags.dm [hunk 4]
```diff
@@ -16,7 +16,8 @@
 	worn_icon_state = "emag"
 	var/prox_check = TRUE //If the emag requires you to be in range
 	var/type_blacklist //List of types that require a specialized emag
-
+	var/pshoom = 'sound/items/pshoom.ogg'
+	var/pshoom_2 = 'sound/items/pshoom_2.ogg'
```
Comment (Mothblocks) [inline code/game/objects/items/emags.dm:20]: These vars seem useless, I'd just inline them

File: code/game/objects/items/emags.dm [hunk 5]
```diff
@@ -64,8 +65,21 @@
 	return ITEM_INTERACT_SUCCESS
 
 /obj/item/card/emag/ranged_interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	if (can_emag(interacting_with, user))
+		if(pshoom_2 && prob(1))
+			playsound(src, pshoom_2, 40, TRUE)
+		else
+			playsound(src, pshoom, 40, TRUE)
```
Comment (Mothblocks) [inline code/game/objects/items/emags.dm:72]: Add named vars for vol and vary

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Put it in the crafting menu too--a slapcraft is fine but the crafting menu is how we can help communicate the things that exist in the world

**Comment by Mothblocks** [comment]
Btw just as a reminder, our policy is generally that it's ok to have OP stuff behind anomaly cores. There's still a limit there, there always is, but we're much more willing to let them live for a bit first

**Comment by DaCoolBoss** [comment]
This should cost extra telecrystals.

**Comment by SomeoneYouProbablyKnow** [comment]
I think it doesn't have to single you out. Something like blue sparks in the thing you are hacking is good enough. It's some feedback but not a giant red (or in this case blue) arrow pointing at you

**Comment by LucyGrind** [comment]
> I think it doesn't have to single you out. Something like blue sparks in the thing you are hacking is good enough. It's some feedback but not a giant red (or in this case blue) arrow pointing at you

I like this idea, I'll see what i can do

File: code/game/objects/items/emags.dm (hunk 1 above)
Comment (LucyGrind) [reply code/game/objects/items/emags.dm:76]: yeah you're right thats weird, forgot to add the return on this

File: code/game/objects/items/emags.dm (hunk 3 above)
Comment (LucyGrind) [reply code/game/objects/items/emags.dm:78]: if i qdel(src) at the end, it'll try to place an object in the active hand before deleting the src item (the emag), no?

File: code/game/objects/items/emags.dm (hunk 4 above)
Comment (LucyGrind) [reply code/game/objects/items/emags.dm:20]: fair point, will do

File: code/game/objects/items/emags.dm (hunk 1 above)
Comment (LucyGrind) [reply code/game/objects/items/emags.dm:76]: fixed it

File: code/game/objects/items/emags.dm (hunk 4 above)
Comment (LucyGrind) [reply code/game/objects/items/emags.dm:20]: What do you mean by this?

**Comment by CCC23** [comment]
> I've messed around with them on a solo server in the past and bluespace emags are hilariously OP. You can use a camera console and hack borgs while they are getting repaired or new modules, randomly explode every single light on the station with 0 risk to yourself, hack arcade games WHILE PEOPLE ARE PLAYING THEM to kill them, hijack the emergency shuttle without even entering the cockpit, set air alarms to siphon and much more.
> 
> It's absolutely wild what you can do with a bluespace emag and there's a really good reason they are an admin only item.

hehehe :)

**Comment by adamkad1** [comment]
> I've messed around with them on a solo server in the past and bluespace emags are hilariously OP. You can use a camera console and hack borgs while they are getting repaired or new modules, randomly explode every single light on the station with 0 risk to yourself, hack arcade games WHILE PEOPLE ARE PLAYING THEM to kill them, hijack the emergency shuttle without even entering the cockpit, set air alarms to siphon and much more.
> 
> It's absolutely wild what you can do with a bluespace emag and there's a really good reason they are an admin only item.

I mean, we got a instakill rifle now, so eh.

**Pull request closed without merging**